ert_shared/models/iterated_ensemble_smoother.py

```diff
@@ -79,7 +79,9 @@ def runSimulations(self, arguments):
         previous_ensemble_name = None
         while current_iter < ERT.enkf_facade.get_number_of_iterations() and num_retries < num_retries_per_iteration:
             pre_analysis_iter_num = analysis_module.getInt("ITER")
-            self.analyzeStep( run_context )
+            if current_iter == 0:
```

Comment:
Would this fit better inside the `analyzeStep` function together with the other hook calls?

Reply:
Maybe, though the iteration number is less explicitly available inside the `analyzeStep`. It also makes it even harder to write the test, which is not an argument in itself, but Occam`s Razor and all.

Reply:
You are right. Let us leave it as is as there is a refactoring coming this way anyway. But, this is the kind of code that might deserve a comment...